components/user-profile/user-modal/index.tsx

```diff
@@ -1,20 +1,59 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import { Button, Col, Form } from "react-bootstrap";
 import { MdAccountBox, MdEmail } from "react-icons/md";
 import styles from "./user-modal.module.css";
+import { updateUserData, getUserData } from "src/actions/user-details";
 
 interface UserProps {
-  userId?: number;
-  userName: string;
-  userEmail: string;
   onButtonSelect: () => void;
 }
 
-const UserForm: React.FunctionComponent<UserProps> = ({
-  onButtonSelect,
-  userName,
-  userEmail,
-}) => {
+const UserForm: React.FunctionComponent<UserProps> = ({ onButtonSelect }) => {
+  const [userDetail, setUserDetail] = useState({
+    name: "",
+    email: "",
+  });
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+    getUserData()
+      .then((data) => {
+        setUserDetail({
+          name: data["name"],
+          email: data["email"],
+        });
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  }, []);
+
+  const getParsedUserDetails = () => {
+    return {
+      ...userDetail,
+    };
+  };
+
+  const updateUserDetails = (event) => {
+    const { name, value } = event.target;
+    setUserDetail({ ...userDetail, [name]: value });
+  };
+
+  const updateEditedUserData = async () => {
+    const newUserDetails = getParsedUserDetails();
+    try {
+      setLoading(true);
+      await updateUserData(newUserDetails);
```

Comment:
Can directly reference `userDetail`. No need to retrieve that info using a separate function since it is returning the same thing.